astropy/table/table.py

```diff
@@ -3021,8 +3048,70 @@ def __eq__(self, other):
 
         return result
 
-    def __ne__(self, other):
-        return ~self.__eq__(other)
+    def cols_equal(self, other):
+        """
+        Element-wise comparison of table with another table, list, or scalar.
+
+        Returns a ``Table`` with the same columns containing boolean values
+        showing result of comparison.
+
+        Parameters
+        ----------
+        other : Table-like object or list or scalar
+             Object to compare with table
+
+        Examples
+        --------
+        Compare one Table with other::
+
+          >>> t1 = Table([[1, 2], [4, 5], [-7, 8]], names=('a', 'b', 'c'))
+          >>> t2 = Table([[1, 2], [-4, 5], [7, 8]], names=('a', 'b', 'c'))
+          >>> t1.cols_equal(t2)
+          <Table length=2>
+           a     b     c
+          bool  bool  bool
+          ---- ----- -----
+          True False False
+          True  True  True
+
+        """
+        # Check the length of other
+        check_len = True
+
+        if isinstance(other, Table):
+            names = other.colnames
+        else:
+            try:
+                other = Table(other, copy=False)
+                names = other.colnames
+            except Exception:
+                # Broadcast other into a dict, so e.g. other = 2 will turn into
+                # other = {'a': 2, 'b': 2} and then equality does a
+                # column-by-column broadcasting.
+                names = self.colnames
+                other = {name: other for name in names}
+                check_len = False
+
+        if check_len and len(self) != len(other):
```

Comment:
Maybe one always wants to allow broadcasting, i.e., either length unity would be OK? That seems more consistent with also allowing a numeric value above.

Reply:
Yes, I think this might make sense, so you can do e.g. `t.values_equal(t[1])` to see if all rows are the same as row 1.  The downside would be getting this darn numpy future warning and no really good diagnosis in the case of comparing a table of length 2 and a table of length 3.  So maybe check the lengths but only allow 1 or len(self)?  Those are the only lengths that can broadcast onto the `self`.

Reply:
Yes, makes sense, just allow those two.